### PR TITLE
Convert to noarch

### PR DESCRIPTION
**Pull request by dbast**
This feedstock currently has 33 builds for various python versions and platforms osx-64/osx-arm64/win-64/linux-{64, aarch64, ppc64le}, but it just contains not-compiled python code which can be handled by a single noarch package (that would then be also installable on new platforms like linux-s390x). 

The package can be made noarch by:
* removing `importlib_resources <5.3  # [py<37]` as it anyways depends on `python >=3.7`
* removing the `# [py<38]` selector from `importlib_metadata` as is not required but still also available for `[py>=3.8]`.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by dbast** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -30,8 +30,7 @@ outputs:
       run:
         - cfgv >=2.0.0
         - identify >=1.0.0
-        - importlib_metadata  # [py<38]
-        - importlib_resources <5.3  # [py<37]
```
Comment (asottile) [inline recipe/meta.yaml:34]: the version specifier got removed here

it might make sense to just make the feedstock require 3.8+ instead ?

Comment (asottile) [reply recipe/meta.yaml:34]: I'm the pre-commit maintainer

there is harm, the latest version of `importlib-resources` subtly breaks pre-commit

Comment (dbast) [reply recipe/meta.yaml:34]: thanks... that is good input! ... hm, then either the feedstock can require python 3.8+ or disallow specific importlib-resources version ranges

Comment (asottile) [reply recipe/meta.yaml:34]: oh wait I just misread this entirely -- this is fine as is

**Comment by dbast** [comment]
@conda-forge-admin, please rerender

**Review by asottile — APPROVED** [review]

**Comment by nicoddemus** [comment]
Hey @dbast, sorry for missing this PR, just noticed it now.

v3.0.0 has been officially released and is about to be published by #58. It drops support for Python <3.8, which enables it no longer use conditional flags anymore, making it an ideal candidate for a `noarch` package.

Would you like to open an equivalent PR based off 3.0.0?

**Pull request closed without merging**